src/lib/request.js

```diff
@@ -32,102 +56,71 @@ export default function (relativeUrl, opts, callback) {
-            }
-        }
-    } else if (typeof(requestedTemplate) === 'string') {
-        templates = [requestedTemplate];
-    } else if (Array.isArray(requestedTemplate) && requestedTemplate.length > 0) {
-        templates = requestedTemplate;
-    }
-
-    if (templates.length > 0) {
-        usingTemplates = true;
-
+    if (usingTemplates) {
         headers['stencil-options'] = JSON.stringify({
             render_with: templates.join(','),
         });
     }
 
-    let url = relativeUrl;
-    if (options.requestOptions.baseUrl) {
-        url = `${options.requestOptions.baseUrl}${url}`;
-    }
+    const url = options.requestOptions.baseUrl ? `${options.requestOptions.baseUrl}${relativeUrl}` : relativeUrl;
 
-    // make ajax request using jquery
-    $.ajax({
+    fetch(url, {
         method: options.method,
-        url,
-        xhrFields: {
-            withCredentials: true,
-        },
-        contentType: options.requestOptions.formData ? false : 'application/x-www-form-urlencoded; charset=UTF-8',
-        processData: !options.requestOptions.formData,
-        success: (response) => {
-            let ret;
-            const content = options.remote ? response.content : response;
-
-            if (usingTemplates) {
-                // Remove the `components` prefix from the response if it's an object
-                if (typeof(content) === 'object') {
-                    const keys = Object.keys(content);
-                    let key;
-
-                    for (key of keys) {
-                        const cleanKey = key.replace(/^components\//, '');
-
-                        content[cleanKey] = content[key];
-                        delete(content[key]);
-                    }
-                }
-
-                // If using "sections", morph the content into the arbitrary keys => content object.
-                if (usingSections) {
-                    const templateVariableNames = Object.keys(requestedTemplate);
-                    let templateVariable;
-                    for (templateVariable of templateVariableNames) {
-                        content[templateVariable] = content[requestedTemplate[templateVariable]];
-                        delete content[requestedTemplate[templateVariable]];
-                    }
+        headers,
+        credentials: 'include',
+        body: !options.requestOptions.formData ? stringify(data) : data,
+    }).then(response => {
+        if (response.headers.get('content-type') === 'application/json') {
+            return response.json();
+        }
+        return response.text();
+    }).then(response => {
+        const content = options.remote ? response.content : response;
+        let ret;
+
+        if (usingTemplates) {
+            // Remove the `components` prefix from the response if it's an object
+            if (typeof(content) === 'object') {
+                const keys = Object.keys(content);
+                let key;
```

Comment:
Seems like this declaration is redundant, we can just move these variables inside for():
`for (let key of Object.keys(content)) {`

Reply:
Agree